app/assets/javascripts/directives/form_changed.js

```diff
@@ -0,0 +1,44 @@
+ManageIQ.angular.app.directive('formChanged', function() {
+  return {
+    require: 'form',
+    link: function(scope, _elem, _attr, ctrl) {
+      var model = function() {
+        return scope.$eval(ctrl.model || scope.model);
+      };
+
+      var modelCopy = function() {
+        return scope.$eval(ctrl.modelCopy || 'modelCopy');
+      };
+
+      var empty = function(object) {
+        return object === undefined || object === '' || object === null;
+      };
+
+      var compare = function(original, copy, key) {
+        // add missing keys in copy from original so recursion works
+        if (_.isObject(copy) && _.isObject(original)) {
+          _.difference(Object.keys(original), Object.keys(copy)).forEach(function(key) {
```

Comment:
Noticed this CC error on this line that can be easily fixed:
`'key' is already declared in the upper scope. `